### PR TITLE
Privacy Pro settings updates

### PR DESCRIPTION
**Pull request by federicocappelli**
Task/Issue URL: https://app.asana.com/0/1205842942115003/1207746442205594/f
CC: @miasma13 

**Description**:

New Subscription settings menu states and new subscription setting pages.

**Steps to test this PR**:
1. All regular Subscription tests for all possible Subscription states (unsubscribed, subscribed, expired, no entitlements)

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [x] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [x] Portrait
* [x] Landscape

**Device Testing**:

* [x] iPhone 15 Pro max
* [x] iPad

**OS Testing**:

* [x] iOS 17

**Theme Testing**:

* [x] Light theme
* [x] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -21,16 +21,6 @@ opt_in_rules:
   - fatal_error_message
   - file_header
 
-custom_rules:
```
Comment (federicocappelli) [inline .swiftlint.yml:24]: removed in order to enable proper logging for subscription, logging will be refactored in https://app.asana.com/0/1205842942115003/1206750146033742/f anyway

File: DuckDuckGo/SettingsSubscriptionView.swift [hunk 2]
```diff
@@ -87,148 +61,157 @@ struct SettingsSubscriptionView: View {
...
-                label: UserText.settingsPProVPNTitle,
-                subtitle: viewModel.state.networkProtection.status != ""
-                ? viewModel.state.networkProtection.status : nil,
-                action: { viewModel.presentLegacyView(.netP) },
-                disclosureIndicator: true,
-                isButton: true)
+            SettingsCellView(label: UserText.settingsPProVPNTitle,
+                             image: Image("SettingsPrivacyProVPN"),
+                             action: { viewModel.presentLegacyView(.netP) },
+                             statusIndicator: StatusIndicatorView(status: viewModel.state.networkProtection.enabled ? .on : .off),
```
Comment (miasma13) [inline DuckDuckGo/SettingsSubscriptionView.swift]: You use `viewModel.state.networkProtection.enabled` but it is never set anywhere in the code. 
Please see `func updateNetPStatus(connectionStatus: ConnectionStatus)` in `SettingsViewModel` (there is only `status` string set there which is obsolete.

Comment (miasma13) [reply DuckDuckGo/SettingsSubscriptionView.swift]: Also if you set `viewModel.state.networkProtection.enabled` to `true` you will get an additional `VPN` cell in the "Privacy Protections" section of the settings.

Comment (federicocappelli) [reply DuckDuckGo/SettingsSubscriptionView.swift]: Fixed, I'm creating a new build

File: DuckDuckGo/SettingsViewModel.swift [hunk 3]
```diff
@@ -461,12 +457,12 @@ extension SettingsViewModel {
         if AppDependencyProvider.shared.vpnFeatureVisibility.isPrivacyProLaunched() {
             switch connectionStatus {
             case .connected:
-                self.state.networkProtection.status = UserText.netPCellConnected
+                self.state.networkProtectionConnected = true
             default:
-                self.state.networkProtection.status = UserText.netPCellDisconnected
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift]: Both `UserText.netPCellConnected` and `UserText.netPCellDisconnected` are not used anywhere else and can be removed from `UserText`.

Comment (federicocappelli) [reply DuckDuckGo/SettingsViewModel.swift]: removed

**Review by miasma13 — APPROVED** [review]
LGTM 👍

**Pull request merged**